baybe/constraints/discrete.py

```diff
@@ -41,6 +44,37 @@ def get_invalid(self, data: pd.DataFrame) -> pd.Index:  # noqa: D102
         res = reduce(_valid_logic_combiners[self.combiner], satisfied)
         return data.index[res]
 
+    def get_polars_expression(self, df: pl.LazyFrame) -> pl.Expr:
+        """Translate the conditions to polars expression for filtering.
+
+        Args:
+            df: A polars lazy frame of all parameters.
+
+        Returns:
+            The Polars Expr object to pass as an argument to filter().
+        """
+        satisfied: list[pl.Expr] = []
+        # Make sure all condition's parameters all present before continuing
+        if all(p in df.columns for p in self.parameters):
+            for k, cond in enumerate(self.conditions):
+                # SubSelectionCondition requires parameter to generate an expression
+                if isinstance(cond, SubSelectionCondition):
+                    satisfied.append(cond.convert_to_polars(self.parameters[k]))
+                else:  # Assuming we have only two classes for condition
+                    op = cond.convert_to_polars(param=None)
+                    satisfied.append(op(pl.col(self.parameters[k])))
+
+        if satisfied:
+            # to prevent errors when we have empty condition list
+            expr = reduce(_valid_logic_combiners[self.combiner], satisfied)
+            # Negate the expression, because Polars' filter keeps the rows
+            # the expression satisfies
+            expr = expr.not_()
+
+            return expr
+
+        return pl.lit(True)
```

Comment:
I removed all these checks because we no longer filter during join (as you described [here](https://github.com/emdgroup/baybe/pull/252#discussion_r1646679946))
 `return pl.lit(True)` is not required anymore.

Reply:
then i dont understand why its still here in the code
is this PR up to date?

Reply:
Sorry, PR was not up to date.